components/status/StatusPreviewCard.vue

```diff
@@ -2,20 +2,23 @@
 import type { mastodon } from 'masto'
 
 const props = defineProps<{
+  status?: mastodon.v1.Status
   card: mastodon.v1.PreviewCard
   /** For the preview image, only the small image mode is displayed */
   smallPictureOnly?: boolean
   /** When it is root card in the list, not appear as a child card */
   root?: boolean
+  linkToStatus?: URL
 }>()
 
-const providerName = props.card.providerName
+const providerName = $computed(() => props.card.providerName ? props.card.providerName : new URL(props.card.url).hostname)
 
 const gitHubCards = $(usePreferences('experimentalGitHubCards'))
 </script>
 
 <template>
-  <LazyStatusPreviewGitHub v-if="gitHubCards && providerName === 'GitHub'" :card="card" />
-  <LazyStatusPreviewStackBlitz v-else-if="gitHubCards && providerName === 'StackBlitz'" :card="card" :small-picture-only="smallPictureOnly" :root="root" />
+  <StatusPreviewMastodon v-if="linkToStatus" :link-to-status="linkToStatus" :source-status="status" :card="card" />
+  <LazyStatusPreviewGitHub v-else-if="gitHubCards && providerName.toLowerCase().includes('github')" :card="card" />
+  <LazyStatusPreviewStackBlitz v-else-if="gitHubCards && providerName.toLowerCase().includes('stackblitz')" :card="card" :small-picture-only="smallPictureOnly" :root="root" />
```

Comment:
I didn't make any change with regard to lazy vs. strict. The changes I made were to: 
- [X] Add the `StatusPreviewMastodon` component
- [X] clean up the business logic to make sure that we're capturing spelling variations or other changes that happen at the source. For example, if `Github` changes their provider name to `GitHub` or `github`, the existing logic would fail.

Reply:
Ah, gotcha. I thought you were referring to the Lazy loading. I'll make the changes.